src/main/java/com/puppycrawl/tools/checkstyle/filters/IntMatchFilterElement.java

```diff
@@ -47,12 +45,12 @@ public String toString() {
     }
 
     @Override
-    public final int hashCode() {
+    public int hashCode() {
         return Integer.valueOf(matchValue).hashCode();
     }
 
     @Override
-    public final boolean equals(Object object) {
+    public boolean equals(Object object) {
```

Comment:
in other class/record you remove such methods.
What is value to keep this methods?

try to look at git history why such methods were added.